[wasm][build] "CreateWasmBundle" task failed unexpectedly

The file that is not found is `dotnet.timezones.blat`

When executing the following command from the documentation [Building the WebAssembly runtime files](https://github.com/dotnet/runtime/blob/master/docs/workflow/building/libraries/webassembly-instructions.md#building-the-webassembly-runtime-files)

```
./dotnet.sh build /p:Configuration=Debug /p:TargetArchitecture=wasm /p:TargetOS=Browser src/libraries/src.proj /t:BuildWasmRuntimes
```

The following error is produced:

```

/Users/Jimmy/websharp/projects/runtime-work/src/mono/wasm/wasm.targets(51,5): error MSB4018: The "CreateWasmBundle" task failed unexpectedly. [/Users/Jimmy/websharp/projects/runtime-work/src/libraries/src.proj]
/Users/Jimmy/websharp/projects/runtime-work/src/mono/wasm/wasm.targets(51,5): error MSB4018: System.IO.DirectoryNotFoundException: Could not find a part of the path '/Users/Jimmy/websharp/projects/runtime-work/artifacts/bin/native/-Browser-Debug-wasm/dotnet.timezones.blat'. [/Users/Jimmy/websharp/projects/runtime-work/src/libraries/src.proj]
/Users/Jimmy/websharp/projects/runtime-work/src/mono/wasm/wasm.targets(51,5): error MSB4018:    at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter) [/Users/Jimmy/websharp/projects/runtime-work/src/libraries/src.proj]
/Users/Jimmy/websharp/projects/runtime-work/src/mono/wasm/wasm.targets(51,5): error MSB4018:    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode) [/Users/Jimmy/websharp/projects/runtime-work/src/libraries/src.proj]
/Users/Jimmy/websharp/projects/runtime-work/src/mono/wasm/wasm.targets(51,5): error MSB4018:    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) [/Users/Jimmy/websharp/projects/runtime-work/src/libraries/src.proj]
/Users/Jimmy/websharp/projects/runtime-work/src/mono/wasm/wasm.targets(51,5): error MSB4018:    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share) [/Users/Jimmy/websharp/projects/runtime-work/src/libraries/src.proj]
/Users/Jimmy/websharp/projects/runtime-work/src/mono/wasm/wasm.targets(51,5): error MSB4018:    at System.IO.File.Open(String path, FileMode mode, FileAccess access, FileShare share) [/Users/Jimmy/websharp/projects/runtime-work/src/libraries/src.proj]
/Users/Jimmy/websharp/projects/runtime-work/src/mono/wasm/wasm.targets(51,5): error MSB4018:    at CreateWasmBundle.Execute() in /Users/Jimmy/websharp/projects/runtime-work/tools-local/tasks/mobile.tasks/CreateWasmBundle/CreateWasmBundle.cs:line 55 [/Users/Jimmy/websharp/projects/runtime-work/src/libraries/src.proj]
/Users/Jimmy/websharp/projects/runtime-work/src/mono/wasm/wasm.targets(51,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/Users/Jimmy/websharp/projects/runtime-work/src/libraries/src.proj]
/Users/Jimmy/websharp/projects/runtime-work/src/mono/wasm/wasm.targets(51,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/Users/Jimmy/websharp/projects/runtime-work/src/libraries/src.proj]
```


Adding the `/p:BuildTargetFramework=net5.0` seems to work.

```
./dotnet.sh build /p:Configuration=Debug /p:TargetArchitecture=wasm /p:TargetOS=Browser /p:BuildTargetFramework=net5.0 src/libraries/src.proj /t:BuildWasmRuntimes
```

After this then the `/p:BuildTargetFramework=net5.0` is no longer required.

--- a/src/libraries/Directory.Build.props
+++ b/src/libraries/Directory.Build.props
@@ -71,7 +71,9 @@
   <!-- Common repo directories -->
   <PropertyGroup>
     <!-- Need to try and keep the same logic as the native builds as we need this for packaging -->
-    <NativeBinDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'native', '$(BuildTargetFramework)-$(TargetOS)-$(Configuration)-$(TargetArchitecture)'))</NativeBinDir>
+    <_targetFrameworkValue>$([MSBuild]::ValueOrDefault('$(BuildTargetFramework)', '$(TargetFramework)'))</_targetFrameworkValue>
+    <_targetFrameworkValue>$([MSBuild]::ValueOrDefault('$(_targetFrameworkValue)', '$(NetCoreAppCurrent)'))</_targetFrameworkValue>
+    <NativeBinDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'native', '$(_targetFrameworkValue)-$(TargetOS)-$(Configuration)-$(TargetArchitecture)'))</NativeBinDir>
     <PkgDir>$([MSBuild]::NormalizeDirectory('$(LibrariesProjectRoot)', 'pkg'))</PkgDir>
   </PropertyGroup>
 

